packages/core/lib/commands/test/run.js

```diff
@@ -3,13 +3,22 @@ const parseCommandLineFlags = options => {
   const grep = options.grep || options.g;
   const bail = options.bail || options.b;
   const reporter = options.reporter || options.r;
-  return {
-    mocha: {
-      grep,
-      bail,
-      reporter
-    }
-  };
+  if (reporter === undefined) {
+    return {
+      mocha: {
+        grep,
+        bail
+      }
+    };
+  } else {
```

Comment:
Just a small point that the else clause of this isn't necessary because both branches return.

Reply:
@sukanyaparashar explained above that actually this is necessary?

Reply:
I meant the second return doesn't need to be enclosed in a redundant else clause. Sorry I wasn't clear and it is a minor point.

Reply:
Just speaking for myself here, I think the explicit `else` here is clearer...

Reply:
@cds-amal This explicit else is necessary because when the reporter is not specified as the command line argument (its is undefined), it should fall back to the truffle-config.js file specification and it happens only when the reporter is not returned. If we return the reporter as undefined, then it displays the default mocha reporter (which is "spec") and does not fall back to the truffle-config.js file.

Reply:
Oh! I misunderstood what you said @cds-amal . Yes, I also agree that both the styles work but else may make the code more clearer to understand.

Reply:
It's your choice, stylistically.